app/controllers/links_controller.rb

```diff
@@ -46,6 +54,11 @@ def load_dependencies
     @link = Link.retrieve(params)
   end
 
+  def redirect_on_failed_validation
+    flash[:link_url] = params[:link][:url].strip
```

Comment:
There's some duplication of `params[:link][:url].strip` - could this be extracted into a method?